synapse/lib/ast.py

```diff
@@ -36,13 +36,30 @@ class AstNode:
     '''
     Base class for all nodes in the Storm abstract syntax tree.
     '''
-    def __init__(self, kids=()):
+    # set to True if recursive runt-safety checks should *not* recurse
+    # into children of this node.
+    runtopaque = False
+
+    def __init__(self, astinfo, kids=()):
         self.kids = []
+        self.astinfo = astinfo
         self.hasast = {}
-        self.lines = (-1, -1)
-        self.textpos = (-1, -1)
         [self.addKid(k) for k in kids]
 
+    def getAstText(self):
+        return self.astinfo.text[self.astinfo.soff:self.astinfo.eoff]
+
+    def getPosInfo(self):
+        return {
+            'lines': (self.astinfo.sline, self.astinfo.eline),
```

Comment:
badsyntaxerror in the parser.py file includes the texthash. do we no longer have that value here?